smslibrary/src/main/java/com/eis/communication/Header.java

```diff
@@ -0,0 +1,90 @@
+package com.eis.communication;
+
+/**
+ * Header abstract class
+ * @author Mattia Fanan
+ *
+ * @param <P> Peer data-type
+ * @param <D> Payload data data-type
+ */
+public abstract class Header<P extends Peer, D> {
+    protected P sourcePeer;
+    protected P destinationPeer;
+    protected boolean canHaveNullPeer;
+
+    /**
+     * @return Header's data
+     */
+    public abstract D getStamp();
+
+    /**
+     * @return source peer in the Header
+     */
+    public P getSourcePeer() {
+        return sourcePeer;
+    }
+
+    /**
+     * @return destination peer in the Header
+     */
+    public P getDestinationPeer() {
+        return destinationPeer;
+    }
+
+    /**
+     * Set source peer in the Header
+     * @param sourcePeer to set in the Header
+     * @return true if sourcePeer is valid, false if sourcePeer is not valid
+     */
+    public boolean setSourcePeer(P sourcePeer) {
+
+        if (sourcePeer == null){
+            //peers can't be both null
+            if(destinationPeer==null)return false;
+
+            if(!canHaveNullPeer)return false;
+
+            this.sourcePeer=sourcePeer;
+            return true;
+        }
+
+        if (sourcePeer.isValid()) {
+            this.sourcePeer = sourcePeer;
+            return true;
+        }
+
+
+        return false;
+    }
+
+    /**
+     * Set destination peer in the Header
+     * @param destinationPeer to set in the Header
+     * @return true if destinationPeer is valid, false if destinationPeer is not valid
+     */
+    public boolean setDestinationPeer(P destinationPeer){
+        if (destinationPeer == null){
+            //peers can't be both null
```

Comment:
What's the point of having two peers then? Can't you just have one and treat it as the sender/receiver?

Reply:
👍
Maybe in future I can delete one peer but now I’m not much sure is the right thing to do